go/patterns-use-cases/src/cron/cron.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	restate "github.com/restatedev/sdk-go"
+	"github.com/robfig/cron/v3"
+)
+
+// JobRequest represents the structure for creating a cron job
+type JobRequest struct {
+	CronExpression string `json:"cronExpression"` // The cron expression e.g. "0 0 * * *" (every day at midnight)
+	Service        string `json:"service"`
+	Method         string `json:"method"`            // Handler to execute with this schedule
+	Key            string `json:"key,omitempty"`     // Optional: Virtual Object key to call
+	Payload        string `json:"payload,omitempty"` // Optional payload to pass to the handler
+}
+
+// JobInfo represents the stored job information
+type JobInfo struct {
+	Req               JobRequest `json:"req"`
+	NextExecutionTime time.Time  `json:"next_execution_time"`
+	NextExecutionID   string     `json:"next_execution_id"`
+}
+
+const JOB_KEY = "job" // Key for storing job information in the Restate object
+
+// CronJobInitiator service for creating new cron jobs
+//
+// A distributed cron service built with Restate that schedules tasks based on cron expressions.
+//
+// Features:
+// - Create cron jobs with standard cron expressions (e.g., "0 0 * * *" for daily at midnight)
+// - Schedule any Restate service handler or virtual object method
+// - Guaranteed execution with Restate's durability
+// - Cancel and inspect running jobs
+//
+// Usage:
+// 1. Send requests to CronJobInitiator.Create() to start new jobs
+// 2. Each job gets a unique ID and runs as a CronJob virtual object
+// 3. Jobs automatically reschedule themselves after each execution
+
+type CronJobInitiator struct{}
+
+func (CronJobInitiator) Create(ctx restate.Context, req JobRequest) (string, error) {
+	jobID := restate.Rand(ctx).UUID().String()
+
+	println("Creating new cron job with ID:", jobID, "for service:", req.Service, "method:", req.Method)
+	job, err := restate.Object[*JobInfo](ctx, "CronJob", jobID, "Initiate").Request(req)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("Job created with ID %s and next execution time %s",
+		jobID, job.NextExecutionTime.Format(time.RFC3339)), nil
+}
+
+type CronJob struct{}
+
+func (CronJob) Initiate(ctx restate.ObjectContext, req JobRequest) (*JobInfo, error) {
+	// Check if job already exists
+	job, err := restate.Get[*JobInfo](ctx, JOB_KEY)
+	if err != nil {
+		return nil, err
+	}
+	if job != nil {
+		return nil, restate.TerminalError(fmt.Errorf("job already exists for this ID"), 500)
+	}
+
+	return scheduleNextExecution(ctx, req)
+}
+
+func (CronJob) Execute(ctx restate.ObjectContext, req JobRequest) error {
+	// Get the job information
+	job, err := restate.Get[*JobInfo](ctx, JOB_KEY)
+	if err != nil {
+		return err
+	}
+	if job == nil {
+		return restate.TerminalError(fmt.Errorf("job not found"), 500)
```

Comment:
There should be a `TerminalErrorf`: https://pkg.go.dev/github.com/restatedev/sdk-go#TerminalErrorf